### PR TITLE
Update LXD Dependency, Add Cluster Module to LXD Server

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch updates the upstream LXD dependency. 

It introduces a cluster module for the new LXD Server, making it cluster-aware, and introduces some test infrastructure to support developing it.

```NewServer``` now includes an error it its return. Tests and call sites are updated accordingly.

The separate ```NewLocalServer``` calls are unified as ```MaybeNewLocalServer```, which returns a nil server if not running on an OS supporting LXD by default.

There are some additional test method renamings for consistency.

## QA steps

- Green unit tests.
- Bootstrapping to LXD and deploying to container machines works.

### CODE REVIEW

File: container/lxd/cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+func (s *imageSuite) TestUseTarget(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	c1Svr := s.NewMockServerClustered(ctrl, "cluster-1")
+	c2Svr := s.NewMockServerClustered(ctrl, "cluster-2")
+
+	c1Svr.EXPECT().UseTarget("cluster-2").Return(c2Svr)
+
+	lxd.NewServer(c1Svr).UseTargetServer("cluster-2")
```
Comment (SimonRichardson) [inline container/lxd/cluster_test.go]: Should there be a test to show what happens if someone attempts to use a server that isn't a valid cluster name?

```go
lxd.NewServer(c1Svr).UseTargetServer("cluster-bad")
```

**Comment by manadart** [comment]
[LXD Simple Local Clustering](https://trello.com/c/anE1s0ml/237-lxd-simple-local-clustering)

File: container/lxd/cluster_test.go (hunk 1 above)
Comment (manadart) [reply container/lxd/cluster_test.go]: I have added such a test.

As per the edited PR description, I had to add an error to the return from ```NewServer```, so there are a few call sites in the diff now.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**